src/control/ros/Conversions.cpp

```diff
@@ -173,8 +281,26 @@ void extractTrajectoryPoint(
 
   Eigen::VectorXd tangentVector;
   auto state = space->createState();
+
+  auto prevState = space->createState();
+  space->convertPositionsToState(previousPoint, prevState);
+
   trajectory->evaluate(timeAbsolute, state);
-  space->logMap(state, tangentVector);
+  space->convertStateToPositions(state, tangentVector);
+
+  auto geodesicInterpolator = std::dynamic_pointer_cast<aikido::statespace::GeodesicInterpolator>(interpolator);
```

Comment:
Why don't we take in a `GeodesicInterpolator`, rather than always dynamic casting it here?

Reply:
1. Consistency: Seems to be the norm to pass in the base class and then cast it to the appropriate one.
2. Flexibility: In case we introduce new interpolator classes and want to use the one that's tied to the trajectory.

I don't have a strong preference, but I'd be inclined to go this route.

Reply:
In that case, can we replace this with methods that are defined in the `Interpolator` class?

Reply:
See below.